streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingTab.scala

```diff
@@ -29,19 +29,30 @@ import StreamingTab._
  */
 private[spark] class StreamingTab(ssc: StreamingContext)
   extends SparkUITab(getSparkUI(ssc), "streaming") with Logging {
-
   val parent = getSparkUI(ssc)
-  val listener = ssc.progressListener
-
+  var listener = ssc.progressListener
   ssc.addStreamingListener(listener)
   attachPage(new StreamingPage(this))
   parent.attachTab(this)
 }
 
-private object StreamingTab {
+private[spark] object StreamingTab {
   def getSparkUI(ssc: StreamingContext): SparkUI = {
     ssc.sc.ui.getOrElse {
       throw new SparkException("Parent SparkUI to attach this tab to not found!")
     }
   }
+  def updateOrCreateStreamingTab(ssc:StreamingContext):Option[StreamingTab]={
```

Comment:
You have some style issues here, like spaces around operators, braces, and parentheses. Have a look at other code. Also we usually write e.g. `foreach { tab =>` not `foreach(tab = {`
I think the code here can be more concise and avoid `return` with a construction built around `getTabs.find(_.isInstanceOf[StreamingTab]).map(...).getOrElse(...)